clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/provider/agent/AmazonInstanceTypeCachingAgent.java

```diff
@@ -103,155 +80,148 @@ public Collection<AgentDataType> getProvidedDataTypes() {
 
   @Override
   public CacheResult loadData(ProviderCache providerCache) {
-    try {
-      Set<String> matchingAccounts =
-          credentialsRepository.getAll().stream()
-              .filter(ac -> ac.getRegions().stream().anyMatch(r -> region.equals(r.getName())))
-              .map(AccountCredentials::getName)
-              .collect(Collectors.toSet());
+    AmazonEC2 amazonEC2 = amazonClientProvider.getAmazonEC2(this.account, this.region);
+    final List<InstanceTypeInfo> instanceTypesInfo = getInstanceTypes(amazonEC2);
 
-      if (matchingAccounts.isEmpty()) {
-        return new DefaultCacheResult(Collections.emptyMap());
-      }
+    Map<String, Collection<CacheData>> cacheResults = new HashMap<>();
 
-      CacheData metadata =
-          providerCache.get(getAgentType(), "metadata", RelationshipCacheFilter.none());
-      MetadataAttributes metadataAttributes = null;
-      if (metadata != null) {
-        metadataAttributes =
-            objectMapper.convertValue(metadata.getAttributes(), MetadataAttributes.class);
-      }
+    // cache instance types for key "metadata" for backwards compatibility
+    Set<String> instanceTypes =
+        instanceTypesInfo.stream()
+            .map(InstanceTypeInfo::getInstanceType)
+            .collect(Collectors.toSet());
+    DefaultCacheData metadata = buildCacheDataForMetadataKey(providerCache, instanceTypes);
+    cacheResults.put(getAgentType(), Collections.singleton(metadata));
+
+    // cache instance types info
+    if (instanceTypesInfo != null && !instanceTypesInfo.isEmpty()) {
```

Comment:
I see some nested `if`s  do you think we can refactor and use early return pattern to make it more readable?
for example
```
    if (instanceTypesInfo == null || instanceTypesInfo.isEmpty()) {
      return new DefaultCacheResult(cacheResults);
    }
```